File: treeView/src/main/java/org/sqlunet/treeview/control/HotQueryController.java
/*
 * Copyright (c) 2023. Bernard Bou
 */

package org.sqlunet.treeview.control;

import org.sqlunet.treeview.R;

/**
 * Query controller (expanding this controller will trigger query)
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
public class HotQueryController extends QueryController
{
	// static private final String TAG = "HotQueryController";

	/**
	 * Constructor
	 *
	 * @param breakExpand whether this controller breaks expansion
	 */
	public HotQueryController(final boolean breakExpand)
	{
		super(breakExpand);
		this.layoutRes = Controller.V2 ? R.layout.layout_query_2 : R.layout.layout_query;
	}

	@Override
	protected void markExpanded()
	{
		this.junctionView.setImageResource(R.drawable.ic_hotquery_expanded);
	}

	@Override
	protected void markCollapsed()
	{
		this.junctionView.setImageResource(R.drawable.ic_hotquery_collapsed);
	}

	@Override
	protected void markDeadend()
	{
		this.junctionView.setImageResource(R.drawable.ic_hotquery_deadend);
	}
}

File: treeView/src/main/java/org/sqlunet/treeview/control/HotQueryController.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.treeview.control

import androidx.annotation.LayoutRes
import org.sqlunet.treeview.R

/**
 * Query controller
 *
 * @param breakExpand whether this controller breaks expansion
 */
open class HotQueryController(breakExpand: Boolean) : BaseController(breakExpand) {

    @LayoutRes
    override val layoutResId = R.layout.layout_node_query

    private var processed = false

    // override fun createNodeView(context: Context, node: TreeNode, minHeight: Int): View {
    //     val view = super.createNodeView(context, node, minHeight)
    //     return view
    // }

    /**
     * Add data to tree by launching the query
     */
    @Synchronized
    fun processQuery() {
        if (!processed) {
            processed = true
            val query = query!!
            query.process(node)
        }
    }

    private val query: Query?
        get() {
            if (node.payload != null) {
                return node.payload!![0] as Query
            }
            return null
        }
}
